codegenerator/cli/templates/static/codegen/src/eventFetching/hyperfuel/HyperFuelSource.res

```diff
@@ -275,11 +240,59 @@ let make = ({chain, endpointUrl}: options): t => {
     let startFetchingBatchTimeRef = Hrtime.makeTimer()
 
     //fetch batch
-    let pageUnsafe = await Helpers.queryLogsPageWithBackoff(
-      () =>
-        HyperFuel.queryLogsPage(~serverUrl=endpointUrl, ~fromBlock, ~toBlock, ~recieptsSelection),
-      logger,
-    )
+    let pageUnsafe = try await HyperFuel.GetLogs.query(
+      ~serverUrl=endpointUrl,
+      ~fromBlock,
+      ~toBlock,
+      ~recieptsSelection,
+    ) catch {
+    | HyperSync.GetLogs.Error(error) =>
+      raise(
+        Source.GetItemsError(
+          Source.FailedGettingItems({
+            exn: %raw(`null`),
+            attemptedToBlock: toBlock->Option.getWithDefault(currentBlockHeight),
+            retry: switch error {
+            | WrongInstance =>
+              let backoffMillis = switch retry {
+              | 0 => 100
+              | _ => 500 * retry
```

Comment:
Maybe this should have ceiling? A multiplicative of 500 is huge if somehow this happens multiple times.

Reply:
There's a global ceiling of 60 seconds

Reply:
Ok sweet 👍🏼